frontend/projects/dipa-frontend/src/app/modules/gantt/gantt.component.ts

```diff
@@ -35,7 +39,22 @@ export class GanttComponent implements OnInit, AfterContentInit {
     .subscribe((data) => {
       this.milestoneData = data;
     });
-    
+
+    for(let i=0; i<this.milestoneData.length; i++) {
+      this.periodStartDateArray.push(this.milestoneData[i].start);
+    }
+    for(let i=0; i<this.taskData.length; i++) {
+      this.periodStartDateArray.push(this.taskData[i].start);
+      this.periodEndDateArray.push(this.taskData[i].end);
+    }
+
+    var earliest = new Date(Math.min.apply(null,this.periodStartDateArray));
+    var last = new Date(Math.max.apply(null,this.periodEndDateArray));
+
+    this.periodStartDate = earliest; 
+    this.periodEndDate = last; 
+
```

Comment:
@kago1988  Diese Berechnung funktioniert so nicht mehr, da die arrays milestoneData und taskData asynchron initialisiert werden. D.h. Du müsstest Dich da eigentlich in die Subscription einhängen (Bzw. abhängiges Observable)